x-pack/plugins/security_solution/public/resolver/store/data/reducer.ts

```diff
@@ -36,6 +38,20 @@ export const dataReducer: Reducer<DataState, ResolverAction> = (state = initialS
       ...state,
       hasError: true,
     };
+  } else if (
+    action.type === 'userRequestedRelatedEventData' ||
+    action.type === 'appDetectedMissingEventData'
+  ) {
+    return {
+      ...state,
+      relatedEventsReady: new Map([...state.relatedEventsReady, [action.payload, false]]),
+    };
+  } else if (action.type === 'serverReturnedRelatedEventData') {
+    return {
+      ...state,
+      relatedEventsReady: new Map([...state.relatedEventsReady, [action.payload.entityID, true]]),
+      relatedEvents: new Map([...state.relatedEvents, [action.payload.entityID, action.payload]]),
```

Comment:
I don't understand the need for two maps:

Any time `relatedEventsReady` has a `true` value, `relatedEvents` will be populated for that same key. Also, `relatedEventsReady` only has a `false` value when the app is waiting for the data to be populated.

Is this missing some logic that indicates a failure to load the data?

Reply:
This has states:
* value for key (in either map) is undefined: not requested
* relatedEventsReady is false: requested, not received
* relatedEventsReady is true: data is received, should be in relatedEvents?